mimic/test/behavior_tests.py

```diff
@@ -0,0 +1,242 @@
+"""
+Automatically generate tests for behavior registration/deletion APIs.
+Also contains helper functions for specific behavior testing.
+"""
+import json
+
+from uuid import UUID, uuid4
+
+from six import text_type, string_types
+
+from twisted.trial.unittest import SynchronousTestCase
+
+from zope.interface import Attribute, Interface
+
+from mimic.test.helpers import json_request, request_with_content
+
+
+class IBehaviorAPITestHelperFactory(Interface):
+    """
+    A class with a `classmethod` that accepts a test case and returns a
+    helper.  This will be used to generate a behavior CRUD test suite.
+    """
+    __name__ = Attribute(
+        "A name which will be used to generate the test suite name.")
+    __module__ = Attribute(
+        "The module in which the test suite should go.")
+
+    def from_test_case(test_case):
+        """
+        A constructor that generates a provider of
+        :class:`IBehaviorAPITestHelper`.
+
+        :param test_case: An instance of a
+            :class:`twisted.trial.unittest.SynchronousTestCase`
+        :return: a :class:`IBehaviorAPITestHelper` provider
+        """
+
+
+class IBehaviorAPITestHelper(Interface):
+    """
+    Set-up and assertion methods required for testing behavior CRUD.
+    :class:`IBehaviorAPITestHelperFactory.from_test_case` will be called at
+    the beginning of every CRUD method, the intent of which is to generate
+    a new instance should this be a class.  If the provider is not a class,
+    then please keep in mind the intended usage.
+
+    The reason there are two interfaces, both clearly intended to be
```

Comment:
There are a couple reasons I think that these should both a single class:

1. Splitting documentation across multiple interfaces when they are coupled by extra requirements makes it harder to get the documentation right.  For example, I don't actually want a  `from_test_case` method, I just want an `__init__` that takes a `test_case`.  Right now I require a docstring to warn people in the interface in both `from_test_case` and in `IBehaviorAPITestHelper` not to keep any extra state around if they chose not to use a single class, because a fresh one needs to be created on every test method, whereas I can just call `__init__` if it were a single class.  I guess as per our conversation, people could do other silly things like store state to a global variable and restore it on every `__init__`, but that seems less likely.

2.  It doesn't seem like there's much use case in the near future for having a factory be separate from the helper.  It seems like splitting off the constructor as a separate object than the thing being constructed, without any concrete examples as to why or what the use case is would make this code harder to understand.

3. Even with the decorator that reduces the boilerplate, documentation is scattered across several interfaces, and it seems like maybe a lot of the documentation for `make_behavior_tests`, et. al. would have to be repeated in `behavior_tests_for_class` if we wanted to make that the friendly function everyone should use.  (e.g. if that is the method we want everyone to use most of the time, that should be the best-documented one of them all, rather than just have links to every other function and interface.)

What you suggest would be a more precise/theoretically correct interface definition though, and moving this comment to the helper decorator and making it the recommended way to generate test cases would be more useful, since this comment doesn't actually make any recommendations.